addons/point_of_sale/models/pos_config.py

```diff
@@ -160,6 +165,11 @@ def _check_company_journal(self):
         if self.journal_id and self.journal_id.company_id.id != self.company_id.id:
             raise UserError(_("The company of the sale journal is different than the one of point of sale"))
 
+    @api.constrains('company_id', 'invoice_journal_id')
+    def _check_company_journal(self):
+        if self.invoice_journal_id and self.invoice_journal_id.company_id.id != self.company_id.id:
+            raise UserError(_("The company of the invoice journal is different than the one of point of sale"))
```

Comment:
"different _from_" is much preferred, especially in simple forms where the two words are next to each other.
And perhaps this direct form avoids the difficulty: "_The invoice journal and sales journal must belong to the same company_" :-)